### PR TITLE
added scoped variants for distinct: distinct_all, distinct_at, distinct_if

### LINKED ISSUE #2948: Add `distinct_at`?
Many of the `_` verb versions have an `_at` option in 0.7.1. Not so for `distinct_`. Could this be added?

### PR DESCRIPTION
**Pull request by romainfrancois**
Closes #2948.

### CODE REVIEW

File: R/colwise-distinct.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#' Select distinct rows by a selection of variables
+#'
+#' These [scoped] variants of [distinct()] extract distinct rows by a
+#' selection of variables. Like [distinct()], you can modify the
```
Comment (krlmlr) [inline R/colwise-distinct.R]: I tend to add a linked reference only for the first mention of a function.

File: R/colwise-distinct.R [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#' variables before ordering with [funs()].
+#'
+#' @inheritParams scoped
+#' @export
+#' @examples
+#' df <- data_frame(x = rep(2:5, each=2) / 2, y = rep(2:3, each = 4) / 2)
```
Comment (krlmlr) [inline R/colwise-distinct.R]: Can you please add a space around `=`? We'll soon have *styler* support for *roxygen2* examples.

File: R/colwise-distinct.R (hunk 1 above)
Comment (romainfrancois) [reply R/colwise-distinct.R]: ✅

File: R/colwise-distinct.R (hunk 2 above)
Comment (romainfrancois) [reply R/colwise-distinct.R]: ✅

**Pull request merged**